### PR TITLE
Signup: Remove Masterbar from Signup

### PR DESCRIPTION
**Pull request by shaunandrews**
The PR aims to simplify the signup flow by removing the Masterbar. The goal is to reduce the height of individual steps in anticipation of the upcoming site previews. Along with removing the Masterbar, I've reduced the font-size, moved the step counter, and moved the back button.

slice

slice 2

There's still a few questions and things to do:

- [ ] When Calypso loads, it shows an empty Masterbar. I think this is ok, but it could be a little confusing. Maybe we can update the loading state to hide the empty Masterbar when loading a signup screen?
- [x] Right now, this PR just hides the Masterbar with CSS, and added some HTML into `signup/main.jsx` — I have a suspicion it would be better to make the `signup__header` into a new component, and not load the Masterbar at all.
- [x] There is sometimes a skip button, but I haven't addressed it yet.
- [x] When on the final loading screen of signup, there's a double WP logo. Maybe we can transition from one to the other, or simply remove one.

### CODE REVIEW

**Comment by sixhours** [comment]
> When Calypso loads, it shows an empty Masterbar. I think this is ok, but it could be a little confusing. Maybe we can update the loading state to hide the empty Masterbar when loading a signup screen?

I've been looking into this, but haven't made any progress; I don't think we have a way to check if we're in signup until after the layout (and therefore Masterbar) loads. We should pull in someone with more Calypso knowledge to confirm for sure, though. @mattwiebe maybe?

**Comment by fditrapani** [comment]
Something to keep in mind as we work through this is how any of these changes impact Jetpack or the importers (/start/import).

**Comment by fditrapani** [comment]
I just tested Jetpack and it seems unaffected. I guess it mostly applies to any work we do on the plans page.

**Comment by sixhours** [comment]
> There is sometimes a skip button, but I haven't addressed it yet.

I haven't found a flow with a Skip button yet; @shaunandrews is there a particular flow you're looking at, or is it that we'd like to incorporate a Skip button at some point?

**Comment by shaunandrews** [comment]
I know we had a skip button at some point. There's a few references in the code around plans: https://github.com/Automattic/wp-calypso/search?q=skip&unscoped_q=skip

The code currently supports a skip, and it seems like something we'd want to support for future work/tests.

The prototype adds a skip option to the vertical selection step:

screen shot 2018-12-10 at 1 57 57 pm

**Comment by sixhours** [comment]
Some thoughts on the placement of a skip button...

Keep all navigation in the same area for consistency:
screen shot 2018-12-10 at 4 03 08 pm

Put it underneath the other content, centered. Less obtrusive, and maybe a better choice if we only let the user skip one or two specific sections:
screen shot 2018-12-10 at 4 04 18 pm

We've also toyed with the idea of removing the step progress indicator entirely. Maybe it replaces that:
screen shot 2018-12-10 at 4 06 20 pm

I personally lean toward centering it under the other content to make it less obvious.

**Comment by shaunandrews** [comment]
Gave this another spin and fixes some issue that came up with the rebase, positioning with the new `siteMockups`, and some overflow issues.

There's still an issue around the domain search and `sticky-panel`:

image 5

File: client/signup/style.scss [hunk 1]
```diff
@@ -3,6 +3,15 @@
...
 		padding: 32px 0 0 0;
 	}
+
+	//Masterbar is hidden but still has height
+	//which is how sticky panel offset is calculated.
+	//Setting height to zero removes the offset
+	//so the sticky panel sticks to the top.
+
+	&.has-no-masterbar .masterbar {
+		height: 0;
```
Comment (shaunandrews) [inline client/signup/style.scss]: @sixhours Does this only affect signup? Should this `height: 0` go in the masterbar.scss file instead?

Comment (sixhours) [reply client/signup/style.scss]: Chatting in Slack, we determined this should stay in signup for now; we don't use this method of hiding the masterbar elsewhere except in Jetpack Connect.

**Comment by shaunandrews** [comment]
@Automattic/onboarding-dev @Automattic/onboarding-design for a review, this feels ready to merge.

**Comment by jblz** [comment]
I just took this for a run on a large display (1600 vertical pixels) for the first time and the content feels kind of "skewed" to the top of the screen for me with lots of "blank" space below the content:

screen shot 2018-12-13 at 12 50 04 pm

Would it be a good idea to center things vertically after some breakpoint or something?

**Comment by mattwiebe** [comment]
Oh, I snuck something in there in https://github.com/Automattic/wp-calypso/pull/28886/commits/13fab53acd24cca4e3f5601646a5a47474bc9c4e to animate out the masterbar when we enter signup. Gives it a less jarring experience.

One flow I notice as a blocker here:

1. With an existing account, use the site switcher to "Add New Site"
2. Choose "Create a shiny new WordPress.com site"
3. When in `/start/`, go back in your browser
4. There's no visible masterbar

I'm not sure where at the controller level we need to dispatch `showMasterbar` to solve that but otherwise this is great and feel free to remove my last commit

**Comment by scruffian** [comment]
maybe we should only hide the masterbar for logged out users?

**Comment by michaeldcain** [comment]
> maybe we should only hide the masterbar for logged out users?

@scruffian I think that since the goal is to reduce the vertical height of the components (to make way for preview), and since this PR adds elements to the space the masterbar used to occupy, it might defeat the purpose.

**Comment by ramonjd** [comment]
> Let's fix the hidden masterbar problem I detail above

👀

**Comment by sixhours** [comment]
> Oh, I snuck something in there in 13fab53 to animate out the masterbar when we enter signup. Gives it a less jarring experience.

So nice!

> I'm not sure where at the controller level we need to dispatch showMasterbar to solve that

Me, neither. After looking at this for a while, I think we're better off using CSS to hide it in signup only; the controller uses CSS, anyway, but it also updates the state. Since we don't hide the masterbar very often, we probably don't need to worry about being able to access its state from elsewhere in the app. Thoughts?

**Comment by ramonjd** [comment]
> I think we're better off using CSS to hide it in signup only; the controller uses CSS, anyway, but it also updates the state. Since we don't hide the masterbar very often, we probably don't need to worry about being able to access its state from elsewhere in the app. Thoughts?

After poking around for a reliable condition under which we can restore `state.ui.masterbarVisibility`  to `true` (is there one?), I'm inclined to concur. 👍 

If we need to manage/toggle the state of the master bar _across_ sign up, all that comes to mind is to add a new property to the signup state tree, e.g., `state.signup.masterbarVisibility`. 🤔

**Comment by fditrapani** [comment]
Just took another look via Calypso live. Overall, this is looking really good with some really minor things can probably be dealt with in a small follow up PR.

image

- The suggestions for the autocomplete are getting cut off.
- The background block on the right seems to be covered by something.
- The animated masterbar feels a little more distracting than when it's static. 

> Is there any case where the master bar should be visible in the signup flow?
I can't see a reason why it would be even for logged in users. @shaunandrews any thoughts?

> If we need to manage/toggle the state of the master bar across sign up, all that comes to mind is to add a new property to the signup state tree
I would imagine we might want to do this at the end when we're about to checkout or transition to Calypso.

**Comment by ramonjd** [comment]
Actually, we _might_ be able to solve with one line in `layout.jsx`...

`masterbarIsHidden: ! masterbarIsVisible( state ) || 'signup' === sectionName,`

**Comment by scruffian** [comment]
> Actually, we might be able to solve with one line in layout/index.jsx...

This looks like the simplest approach.

**Comment by shaunandrews** [comment]
Just tested the latest, and it seems the case of the missing masterbar has been solved.

> The suggestions for the autocomplete are getting cut off.

I'm not able to reproduce this — can you give more details @fditrapani (Irregardless, I don't think this is a blocker.)

> The background block on the right seems to be covered by something.

I don't think this is related to this PR, I see it on staging and production, too.

> Is there any case where the master bar should be visible in the signup flow?

I don't think so, no. Let's continue with the assumption that there is no Masterbar in any signup flow.

**Comment by fditrapani** [comment]
> I'm not able to reproduce this — can you give more details @fditrapani (Irregardless, I don't think this is a blocker.)

Good call. Let's file it as a bug and address it later.

**Comment by mattwiebe** [comment]
This is good as far as I can tell now. I pushed a few fixes:

* ensure masterbar is hidden when entering `/start/` logged-out
* don't cut off the dropdown in the `site-topic` step
* adjust the `pressable-nux` step to not overlap the (W) logo (see screenshots below - I stumbled on this because I went hunting for `.about__pressable-wrapper` and started there and I still have no idea what the heck is going on there )

before:
image

after:
image

**Pull request merged**